### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by pezholio**
This updates the documentation to make the project easier to get to grips with, as well as using the scripts to rule them all pattern across the project and updating some of the CI

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,16 +2,40 @@
 
 # Golden Retriever
 
-Imports opportuntities from the [Digital Marketplace](https://www.digitalmarketplace.service.gov.uk/digital-outcomes-and-specialists/opportunities) into Hubspot
+Golden Retriever is imports opportuntities from the [Digital Marketplace](https://www.digitalmarketplace.service.gov.uk/digital-outcomes-and-specialists/opportunities) into Hubspot and drops an alert into our Slack like this:
```
Comment (dragon-dxw) [inline README.md]: Delete 'is': "Golden Retriever imports opportunities"

File: README.md [hunk 2]
```diff
@@ -2,16 +2,40 @@
...
+```bash
+script/setup
+```
+
+This script installs all the necessary dependencies and sets up all the required
+environment variables detailed in [.env.example](https://github.com/dxw/golden-retriever/blob/master/.env.example), so make sure you have these to hand ([More on environment variables](https://github.com/dxw/golden-retriever#environment-variables)).
+
+## Environment Variables
+
+For the app to run properly, you need a to have the following environment variables
```
Comment (mec) [inline README.md]: Does the app work at all without these? If not I would be templeted to say these are required, something like:

> You must have the following required environment variables set in hte .env file for the application to function as expected:

Might also be worth mentioning that the setup script asks for them?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -11,24 +11,15 @@ on: [push]
...
-        key: ${{ runner.os }}-gem-${{ hashFiles('**/Gemfile.lock') }}
-        restore-keys: |
-          ${{ runner.os }}-gem-
-    - name: Build and test with Rake
-      run: |
-        gem install bundler
-        bundle install --jobs 4 --retry 3 --path ./vendor/bundle
-        bundle exec rake
+      - uses: actions/checkout@v1
+      - uses: ruby/setup-ruby@v1
```
Comment (mec) [inline .github/workflows/build.yml:18]: 👌

File: script/cibuild [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+# script/cibuild: Setup environment for CI to run tests. This is primarily
+#                 designed to run on the continuous integration server.
+
+set -e
+
+cd "$(dirname "$0")/.."
+
+bundle exec rspec
```
Comment (mec) [inline script/cibuild]: Why did the end of this change from `bundle exec rake` to `bundle exec rspec` is `rspec` just the default? I am being lazy and could probably look…

**Review by mec — CHANGES_REQUESTED** [review]
Looks good, one question on the `cibuild` script for my own learning, answer that and I'll give you the ship. :)

File: script/cibuild (hunk 4 above)
Comment (pezholio) [reply script/cibuild]: No reason really. I might as well make it consistent though 👍

Comment (mec) [reply script/cibuild]: I definately prefer to be explicit and avoid the default task and the naked `rake` expecially in scripts? 👌

**Review by mec — APPROVED** [review]

**Pull request merged**